GameOfRunes/GameOfRunes/GameEngine/GameEngine.swift

```
@@ -170,13 +193,47 @@ class GameEngine {
         increasePlayerMana(by: -manaPoints)
     }
     
+    func stopMovement(for enemyEntity: Entity, duration: TimeInterval) {
+        systemDelegate.stopMovement(
+            for: enemyEntity,
+            duration: GameConfig.IcePrisonPowerUp.powerUpDuration
+        )
+    }
+    
     func runFadingAnimation(_ entity: Entity) {
         systemDelegate.runFadingAnimation(entity)
     }
     
     func setLabel(_ entity: Entity, label: String) {
         systemDelegate.setLabel(entity, label: label)
     }
+    
+    func addComboScore(count: Int) {
+        // TODO: Better combo score multiplier system
+        let totalScore = count * GameConfig.Enemy.normalScore
+        addScore(by: totalScore)
+    }
+    
+    func decreaseLabelOpacity(_ entity: Entity) {
+        systemDelegate.decreaseLabelOpacity(entity)
+    }
+    
+    func incrementCombo() {
+        if comboEntity == nil {
+            add(ComboEntity(gameEngine: self))
+        }
+        guard let comboEntity = comboEntity else {
+            return
```

should throw a fatal error here